ui/app/services/stats-trackers-registry.js

```diff
@@ -30,13 +33,17 @@ export default Service.extend({
 
     const type = resource && resource.constructor.modelName;
     const key = `${type}:${resource.get('id')}`;
-
-    const cachedTracker = registry.get(key);
-    if (cachedTracker) return cachedTracker;
-
     const Constructor = type === 'node' ? NodeStatsTracker : AllocationStatsTracker;
     const resourceProp = type === 'node' ? 'node' : 'allocation';
 
+    const cachedTracker = registry.get(key);
+    if (cachedTracker) {
+      // It's possible for the resource on a cachedTracker to have been
+      // deleted. Rebind it if that's the case.
+      if (!exists(cachedTracker, resourceProp)) cachedTracker.set(resourceProp, resource);
```

Comment:
if rebinding doesn't cause any side-effects , 👍

Reply:
It shouldn't. Everything is keyed by model ID, so this is strictly to attach a new ED model for the same nomad resource.